settings.php

```diff
@@ -230,16 +230,24 @@
         saml2_settings::OPTION_DUAL_LOGIN_NO      => get_string('no'),
         saml2_settings::OPTION_DUAL_LOGIN_YES     => get_string('yes'),
         saml2_settings::OPTION_DUAL_LOGIN_PASSIVE => get_string('passivemode', 'auth_saml2'),
+        saml2_settings::OPTION_DUAL_LOGIN_TEST    => get_string('test_idp_conn', 'auth_saml2')
     ];
-    $dualloginoptions = $yesno;
-    $dualloginoptions[] = get_string('passivemode', 'auth_saml2');
     $settings->add(new admin_setting_configselect(
             'auth_saml2/duallogin',
             get_string('duallogin', 'auth_saml2'),
             get_string('duallogin_help', 'auth_saml2'),
             saml2_settings::OPTION_DUAL_LOGIN_YES,
             $dualloginoptions));
 
+    if (get_config('auth_saml2', 'duallogin') == saml2_settings::OPTION_DUAL_LOGIN_TEST) {
```

Comment:
```suggestion
    if (get_config('auth_saml2', 'duallogin') === saml2_settings::OPTION_DUAL_LOGIN_TEST) {
```

Reply:
The == is intentional, incase it comes in as a string type from the get_config call. It *shouldnt* but no reason not to trust it, as nothing else can really coerce to 3